src/supertux/main.cpp

```diff
@@ -265,20 +264,20 @@ std::string olduserdir = FileSystem::join(physfs_userdir, PACKAGE_NAME);
 std::string olduserdir = FileSystem::join(physfs_userdir, "." PACKAGE_NAME);
 #endif
 if (FileSystem::is_directory(olduserdir)) {
-  boost::filesystem::path olduserpath(olduserdir);
-  boost::filesystem::path userpath(userdir);
+  std::filesystem::path olduserpath(olduserdir);
+  std::filesystem::path userpath(userdir);
 
-  boost::filesystem::directory_iterator end_itr;
+  std::filesystem::directory_iterator end_itr;
 
   bool success = true;
 
   // cycle through the directory
-  for (boost::filesystem::directory_iterator itr(olduserpath); itr != end_itr; ++itr) {
+  for (std::filesystem::directory_iterator itr(olduserpath); itr != end_itr; ++itr) {
   try
   {
-    boost::filesystem::rename(itr->path().string().c_str(), userpath / itr->path().filename());
+    std::filesystem::rename(itr->path().string().c_str(), userpath / itr->path().filename());
```

Comment:
Why create a temporary string by calling `string()` and then get a `const char*` instead of just `itr->path.c_str()`?

But why get a `const char*` from it at all, when that just has to be converted to a new `filesystem::path` (which allocates more memory and parses the path again)?

```suggestion
    std::filesystem::rename(itr->path(), userpath / itr->path().filename());
```